jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -0,0 +1,259 @@
+    return this;
+  }
+
+  /**
+   * Adds a layer (defined by a {@link LayerConfiguration}).
+   *
+   * @param layerConfiguration the {@link LayerConfiguration}
+   * @return this
+   */
+  public JibContainerBuilder addLayer(LayerConfiguration layerConfiguration) {
+    layerConfigurations.add(layerConfiguration);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint. This is the beginning of the command that is run when the
+   * container starts. {@link #setProgramArguments} sets additional tokens.
+   *
+   * @param entrypointTokens a list of tokens for the entrypoint command
+   * @return this
+   */
+  public JibContainerBuilder setEntrypoint(List<String> entrypointTokens) {
+    entrypoint = ImmutableList.copyOf(entrypointTokens);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint.
+   *
+   * @param entrypointTokens tokens for the entrypoint command
+   * @return this
+   * @see #setEntrypoint(List) for more details
+   */
+  public JibContainerBuilder setEntrypoint(String... entrypointTokens) {
+    setEntrypoint(Arrays.asList(entrypointTokens));
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint program arguments. These are additional tokens added to the end
+   * of the entrypoint command.
+   *
+   * <p>For example, if the entrypoint was {@code myprogram --flag subcommand} and program arguments
+   * were {@code hello world}, then the command that run when the container starts is {@code
+   * myprogram --flag subcommand hello world}.
+   *
+   * @param programArguments a list of program argument tokens
+   * @return this
+   */
+  public JibContainerBuilder setProgramArguments(List<String> programArguments) {
+    this.programArguments = ImmutableList.copyOf(programArguments);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint program arguments.
+   *
+   * @param programArguments program arguments tokens
+   * @return this
+   * @see #setProgramArguments(List) for more details
+   */
+  public JibContainerBuilder setProgramArguments(String... programArguments) {
+    setProgramArguments(Arrays.asList(programArguments));
+    return this;
+  }
+
+  /**
+   * Sets the container environment. These environment variables are available to the program
+   * launched by the container entrypoint command. This replaces any previously-set environment
+   * variables.
+   *
+   * @param environmentMap a map of environment variable names to values
+   * @return this
+   */
+  public JibContainerBuilder setEnvironment(Map<String, String> environmentMap) {
+    environment = new HashMap<>(environmentMap);
+    return this;
+  }
+
+  /**
+   * Sets a variable in the container environment.
+   *
+   * @param name the environment variable name
+   * @param value the environment variable value
+   * @return this
+   * @see #setEnvironment for more details
+   */
+  public JibContainerBuilder setEnvironmentVariable(String name, String value) {
```

Comment:
`addEnvironmentVariable`?

Reply:
If the environment variable already exists, I think the naming is still accurate.

Reply:
I think we should keep this consistent with `add/setLabel`. Should we change this to `add...` or change `addLabel` to `setLabel`?